src/main/java/org/openstreetmap/atlas/tags/oneway/OneWayTag.java

```diff
@@ -0,0 +1,166 @@
+import org.openstreetmap.atlas.tags.annotations.TagValueDeprecated;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.tags.oneway.bicycle.BicycleOneWayTag;
+import org.openstreetmap.atlas.tags.oneway.bicycle.CyclewayLeftOneWayTag;
+import org.openstreetmap.atlas.tags.oneway.bicycle.CyclewayOneWayTag;
+import org.openstreetmap.atlas.tags.oneway.bicycle.CyclewayRightOneWayTag;
+import org.openstreetmap.atlas.tags.oneway.bicycle.OneWayBicycleTag;
+import org.openstreetmap.atlas.tags.oneway.motor.OneWayMotorVehicleTag;
+import org.openstreetmap.atlas.tags.oneway.motor.OneWayMotorcarTag;
+import org.openstreetmap.atlas.tags.oneway.motor.OneWayVehicleTag;
+
+/**
+ * OSM's oneway tag
+ *
+ * @author cstaylor
+ * @author matthieun
+ */
+@Tag(taginfo = "http://taginfo.openstreetmap.org/keys/oneway#values", osm = "http://wiki.openstreetmap.org/wiki/Key:oneway")
+public enum OneWayTag
+{
+    YES,
+    NO,
+    REVERSIBLE,
+    @TagValueDeprecated
+    TRUE,
+    @TagValueDeprecated
+    FALSE,
+    @TagValueDeprecated
+    @TagValueAs("1")
+    ONE,
+    @TagValueDeprecated
+    @TagValueAs(value = "0")
+    ZERO,
+    @TagValueAs("-1")
+    MINUS_1,
+    @TagValueDeprecated
+    REVERSE;
+
+    @TagKey
+    public static final String KEY = "oneway";
+
+    protected static final Set<OneWayTag> ONE_WAYS_FORWARD = EnumSet.of(YES, TRUE, ONE);
+    // Note here that REVERSIBLE is not reversed.
+    protected static final Set<OneWayTag> ONE_WAYS_REVERSED = EnumSet.of(MINUS_1, REVERSE);
+    protected static final Set<OneWayTag> TWO_WAYS = EnumSet.of(NO, FALSE, ZERO);
+
+    public static boolean isBicycleOneWayForward(final Taggable taggable,
+            final boolean rightHandDrive)
+    {
+        return BicycleOneWayTag.isOneWayForward(taggable)
+                || CyclewayOneWayTag.isOneWayForward(taggable)
+                || OneWayBicycleTag.isOneWayForward(taggable)
+                || (rightHandDrive && CyclewayRightOneWayTag.isOneWayForward(taggable))
+                || (!rightHandDrive && CyclewayLeftOneWayTag.isOneWayForward(taggable));
+    }
+
+    public static boolean isBicycleOneWayReversed(final Taggable taggable,
+            final boolean rightHandDrive)
+    {
+        return BicycleOneWayTag.isOneWayReversed(taggable)
+                || CyclewayOneWayTag.isOneWayReversed(taggable)
+                || OneWayBicycleTag.isOneWayReversed(taggable)
+                || (rightHandDrive && CyclewayRightOneWayTag.isOneWayReversed(taggable))
+                || (!rightHandDrive && CyclewayLeftOneWayTag.isOneWayReversed(taggable));
+    }
+
+    public static boolean isBicycleTwoWay(final Taggable taggable)
+    {
+        return BicycleOneWayTag.isTwoWay(taggable) || CyclewayOneWayTag.isTwoWay(taggable)
+                || OneWayBicycleTag.isTwoWay(taggable) || CyclewayRightOneWayTag.isTwoWay(taggable)
+                || CyclewayLeftOneWayTag.isTwoWay(taggable);
+    }
```

Comment:
Somehow the diff is not like just a rename of the class, but this is the new part here